lib/diagnostic.js

```diff
@@ -6,29 +6,73 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+let normalizedWords = [];
+let getUniqueWords = [];
+
+const normalizeWords = function ( text )
+{
+  let words = text.split(/\s+/);
+  for (let i = 0; i < words.length; i++)
+  {
+    normalizedWords[i] = words[i].replace( /\W+/, '' ).toUpperCase();
+  }
+  return normalizedWords;
 };
 
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
+const uniqueWords = function ( text )
+{
+  let uniqueWordsAsKeys = {};
+  for (let i = 0; i < normalizedWords.length; i++)
+  {
+      uniqueWordsAsKeys[normalizedWords[i]] = true;
+  }
+
+  let keys = [];
+  let k = 0;
+
+
+  for( keys[k++] in uniqueWordsAsKeys ); // jshint ignore:line
+
+  getUniqueWords = keys;
+
+  return getUniqueWords;
+
```

Comment:
Your thinking is on target, but the code is not implemented correctly. From what I can tell, this function only returns the value of `normalizedWords`. Please take a look at the solution for this problem.